src/overlay.py

```diff
@@ -63,6 +65,18 @@ def add_attribute(
         attribute_format_as: type,
     ) -> None:
         """Add an attribute to the overlay. The attribute can be formatted as int, float, or str."""
+
+        if not hasattr(obj, attribute_name):
+            if isinstance(obj, LinearPhysicalObject) and not hasattr(
+                obj.kinematic, attribute_name
```

Comment:
A kinematic should have no name.
Some attributes that we might want to display are not part of the object itself, they are attributes of a structure in the object (i.e. if we want to display the velocity, we can't access it with `obj.velocity`, we have to use `obj.kinematic.velocity`).

I wanted to be able to use the Overlay like:
`overlay.add_attribute(ego, "velocity", "Velocity", int)`
instead of
`overlay.add_attribute(ego.kinematic, "velocity", "Velocity", int)`
or
`overlay.add_attribute(ego, "kinematic.velocity", "Velocity", int)`
or something like this.

Found it easyer to use, but can see that there are downsites as well. 
Leave like this or change in one of the above ways?

Reply:
Ah ok now I get your intention. Indeed  it is more convenient this way and I will approve if you prefer it. However, the overlay class is already somewhat overengineered and this special treatment makes it more difficult to understand and  it is coupled too strongly with implementation details from objects. my personal preference would be the third option. Or, if this turns out to be different,the second one